mlrun/datastore/__init__.py

```diff
@@ -168,15 +169,26 @@ def get_stream_pusher(stream_path: str, **kwargs):
             raise ValueError(f"unsupported stream path {stream_path}")
 
 
-class _DummyStream:
+class _DummyStream(storey.Flow):
     """stream emulator for tests and debug"""
 
     def __init__(self, event_list=None, **kwargs):
         self.event_list = event_list or []
+        super().__init__()
 
     def push(self, data, **kwargs):
         if not isinstance(data, list):
             data = [data]
         for item in data:
             logger.info(f"dummy stream got event: {item}, kwargs={kwargs}")
             self.event_list.append(item)
+
+    async def _do(self, event):
+        if event is storey.flow._termination_obj:
+            return await self._do_downstream(storey.flow._termination_obj)
+        if hasattr(event, "body"):
```

Comment:
Any reason we need to check this here? It's expected that the event will have a body field.

Reply:
I did it before the check on the `_termination_obj`... maybe it can be removed but it safer like that.
ANd as I mention on the next comment it only for simple test flows, not for users.

Reply:
I think in this case it should be easier to define a simple `def do`, not inherit `storey.Flow`, and not handle `storey.flow._termination_obj`. MLRun will wrap it in a `MapClass`. The low level functionality of `storey.Flow` isn't really relevant here.

Reply:
Its not worked to me.. I have to use `storey.Flow` or to add a lot of attribute such as `_legal_first_step`